client/components/login/index.js

```diff
@@ -15,8 +15,9 @@ const mapStateToProps = state => {
 
 const mapDispatchToProps = dispatch => {
   return {
-    authenticate: status => {
+    authenticate: (status, callback) => {
       dispatch({type: SET_AUTHENTICATION_STATUS, payload: status});
+      setTimeout(callback, 0);
```

Comment:
Uses to asyncronously call `callback` function after change of `SET_AUTHENTICATION_STATUS`. This helps to appear `toast` on the page after redirect.